src/partition/repart.c

```diff
@@ -2492,6 +2492,33 @@ static int context_read_definitions(Context *context) {
 
         }
 
+        /* Make sure we won't end up with more than one ESP or XBOOTLDR partition (which could break the
+         * bootloader) */
+
+        Partition *esp = NULL, *xbootldr = NULL;
```

Comment:
You are never dereffing these, hence maybe make these booleans? `esp_seen` and `xbootldr_seen`?

Reply:
No. They get dereffed in subsequent commits. That would be churn for no reason. Realistically the requirement that both definitions exist is a requirement of `PreferESP=`. It's not random

Reply:
The commit (98cb590041d2fc15e39b89cfb0e8ac61236f4f28) is no longer required by the overarching feature so I'm dropping it